src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -346,6 +350,8 @@ protected override void Dispose(bool disposing)
                     Rollback();
                 }
 
+                _connector.Connection?.EndBindingScope(ConnectorBindingScope.Transaction);
```

Comment:
If the user commits and disposes, don't we call EndBindingScope twice? In other words, shouldn't this be moved up inside the !IsCompleted condition above?